packages/client/src/internals/TRPCClient.ts

```diff
@@ -124,34 +123,35 @@ export class TRPCClient<TRouter extends AnyRouter> {
       return () => ({});
     }
 
-    const transformer: DataTransformer = opts.transformer
-      ? 'input' in opts.transformer
-        ? {
-            serialize: opts.transformer.input.serialize,
-            deserialize: opts.transformer.output.deserialize,
-          }
-        : opts.transformer
-      : {
+    function getTransformer(): DataTransformer {
+      if (!opts.transformer)
+        return {
           serialize: (data) => data,
           deserialize: (data) => data,
         };
+      if ('input' in opts.transformer)
+        return {
+          serialize: opts.transformer.input.serialize,
+          deserialize: opts.transformer.output.deserialize,
+        };
+      return opts.transformer;
+    }
 
     this.runtime = {
       AbortController: AC,
       fetch: _fetch,
       headers: getHeadersFn(),
-      transformer,
+      transformer: getTransformer(),
     };
 
-    if ('links' in opts) {
-      this.links = opts.links.map((link) => link(this.runtime));
-    } else {
-      this.links = [
-        httpBatchLink({
-          url: opts.url,
-        })(this.runtime),
-      ];
-    }
+    const getLinks = (): OperationLink<TRouter>[] => {
+      if ('links' in opts) {
```

Comment:
We can do a DX tweak here

```suggestion
We can do a DX tweak here,
```suggestion
        if ('links' in opts) {
          if ('url' in opts) {
            throw new Error('You defined `links`, your url with have no effect');
          }
```

But would be better if it was enforced on a type level 🤔